patches/tModLoader/Terraria/ModLoader/ItemLoader.cs

```diff
@@ -1431,89 +1431,107 @@ public static bool CanStackInWorld(Item increase, Item decrease)
-	/// <param name="decrease">Item where the stack is being reduced.</param>
-	/// <param name="numTransfered">Amount to be transfered </param>
-	/// <param name="infiniteSource">The final stack of item2</param>
-	public static bool TryStackItems(Item increase, Item decrease, out int numTransfered, bool infiniteSource = false)
+	/// <param name="destination">The item instance that <paramref name="source"/> will attempt to stack onto</param>
+	/// <param name="source">The item instance being stacked onto <paramref name="destination"/></param>
+	/// <param name="numTransferred">The quanity of <paramref name="source"/> that was transferred to <paramref name="destination"/></param>
+	/// <param name="infiniteSource">Whether <paramref name="source"/>.stack should be decreased</param>
+	public static bool TryStackItems(Item destination, Item source, out int numTransferred, bool infiniteSource = false)
 	{
-		numTransfered = 0;
+		numTransferred = 0;
 
-		if (!CanStack(increase, decrease))
+		if (!CanStack(destination, source))
 			return false;
 
-		StackItems(increase, decrease, out numTransfered, infiniteSource);
+		StackItems(destination, source, out numTransferred, infiniteSource);
 
 		return true;
 	}
 
 	/// <summary>
-	/// Stacks <paramref name="increase"/> and <paramref name="decrease"/> items.
-	/// <br/> Calls all <see cref="GlobalItem.OnStack"/> and <see cref="ModItem.OnStack"/> hooks if <paramref name="increase"/>.stack is less than <paramref name="increase"/>.maxStack.
+	/// Attempts to stack <paramref name="destination"/> onto <paramref name="source"/>
 	/// </summary>
-	/// <param name="increase">Item where the stack is being increased.</param>
-	/// <param name="decrease">Item where the stack is being reduced.</param>
-	/// <param name="numTransfered">Amount to be transfered </param>
-	/// <param name="infiniteSource"></param>
-	/// <param name="numToTransfer">Used to only transfer a specidied amount instead of all.</param>
-	public static void StackItems(Item increase, Item decrease, out int numTransfered, bool infiniteSource = false, int? numToTransfer = null)
+	/// <param name="destination">The item instance that <paramref name="source"/> will attempt to stack onto</param>
+	/// <param name="source">The item instance being stacked onto <paramref name="destination"/></param>
+	/// <param name="numTransferred">The quanity of <paramref name="source"/> that was transferred to <paramref name="destination"/></param>
+	/// <param name="infiniteSource">Whether <paramref name="source"/>.stack should be decreased</param>
+	/// <param name="numToTransfer">
+	/// An optional argument used to specify the quantity of items to transfer form <paramref name="source"/> to <paramref name="destination"/>.<br/>
+	/// By default, the transfer quantity is the minimum between <paramref name="source"/>.stack and <paramref name="destination"/>.maxStack - <paramref name="destination"/>.stack
```

Comment:
Prefer, `By default, as many items as possible will be transferred. That is, either source will be empty, or destination will be full (maxStack)`